### PR TITLE
Run all benchmarks "as tests" in short setup on every PR

### PR DESCRIPTION
**Pull request by bkontur**
Inspired by: https://github.com/paritytech/cumulus/issues/2806#issuecomment-1618050187

Closes: https://github.com/paritytech/cumulus/issues/2806

**TODOs**
- [x] check ` # TODO: can we remove this?`
- [x] update substrate ref because of `asset-hub-westend` issue

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -142,7 +145,10 @@ include:
   - scripts/ci/gitlab/pipeline/test.yml
   # # build jobs
   - scripts/ci/gitlab/pipeline/build.yml
+  # short-benchmarks jobs
+  - scripts/ci/gitlab/pipeline/short-benchmarks.yml
   # # benchmarks jobs
+  # # TODO: can we remove this?-
   - scripts/ci/gitlab/pipeline/benchmarks.yml
```
Comment (ggwpez) [inline .gitlab-ci.yml:152]: Looks like this one is automatically creating a MR to update the weights? I dont know if that is used.

**Review by ggwpez — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (alvicsam) [reply .gitlab-ci.yml:152]: Correct, but it works only with `release-parachains-v*` branches

File: scripts/ci/gitlab/pipeline/short-benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# This file is part of .gitlab-ci.yml
+# Here are all jobs that are executed during "short-benchmarks" stage
+
+# Run all pallet benchmarks only once to check if there are any errors
+.short-benchmark-template: &short-bench
+  stage: short-benchmarks
+  extends:
+    - .common-refs
+    - .docker-env
+  # this is an artificial job dependency, for pipeline optimization using GitLab's DAGs
```
Comment (alvicsam) [inline scripts/ci/gitlab/pipeline/short-benchmarks.yml:10]:
```suggestion
```
In this case it's not artificial :)

File: .gitlab-ci.yml (hunk 1 above)
Comment (EgorPopelyaev) [reply .gitlab-ci.yml:152]: @ggwpez yes, it is used for the releases, when the benchmark job is triggered from git lab. We used to do it manually so far

File: scripts/ci/gitlab/pipeline/short-benchmarks.yml (hunk 2 above)
Comment (bkontur) [reply scripts/ci/gitlab/pipeline/short-benchmarks.yml:10]: ah, you're right :)

File: .gitlab-ci.yml (hunk 1 above)
Comment (bkontur) [reply .gitlab-ci.yml:152]: @EgorPopelyaev @ggwpez @joepetrowski @chevdor 
so for the very next release to regenerate weights, do we want to use:
1. old way with manual gitlab job?
2. new `bot bench $ all` ?

Comment (EgorPopelyaev) [reply .gitlab-ci.yml:152]: @bkontur lets's proceed with the new `bot` solution if it does a thing, but I would still keep the old pipeline yaml, as a plan B. At least for the next release

Comment (bkontur) [reply .gitlab-ci.yml:152]: @EgorPopelyaev ok, good point

**Comment by bkontur** [comment]
bot rebase

**Review by sergejparity — APPROVED** [review]

**Comment by bkontur** [comment]
bot rebase

**Review by alvicsam — APPROVED** [review]

**Review by EgorPopelyaev — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3186011

**Comment by bkontur** [comment]
bot merge

**Pull request merged**